### ISSUE TITLE
fast/gradients/css3-color-stop-invalid.html - box is green instead of white

### ISSUE THREAD

**Issue by annulen** [qtwebkit/qtwebkit#230]
Culprit is the next code in GradientQt.cpp:

```
qreal stopPosition = qMin(lastStop, qreal(1.0f));
```

Changing it to just `lastStop` fixes the problem, but we need to check if https://bugs.webkit.org/show_bug.cgi?id=41484 reproduces

**Comment by antisergey** [comment]
My problems seems to be quite similar.

There's a linear gradient which has two stops with offset 100%. Element using this gradient is expected to be filled with first stop color, but it's filled with second.

This problem is not reproduced in the latest build of WebKit https://bugs.webkit.org/show_bug.cgi?id=176500 .

I used qtwebkit-5.212.0_alpha2-qt59-msvc2015-x64.zip.

```
<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
<svg xmlns:xlink="http://www.w3.org/1999/xlink" width="800" style="background:ivory" xmlns="http://www.w3.org/2000/svg" height="600" xmlns:svg="http://www.w3.org/2000/svg">
  <defs>   
    <linearGradient x1="0%" spreadMethod="pad" id="lg1000" x2="0%" y2="0%" y1="100%">
      <stop stop-opacity="1" stop-color="red" offset="50%"/>
      <stop stop-opacity="1" stop-color="rgb(255, 255, 255)" offset="50%"/>
    </linearGradient>
	<linearGradient x1="0%" spreadMethod="pad" id="lg1001" x2="0%" y2="0%" y1="100%">
      <stop stop-opacity="1" stop-color="red" offset="100%"/>
      <stop stop-opacity="1" stop-color="rgb(255, 255, 255)" offset="100%"/>
    </linearGradient>
  </defs>
  <g id="layer1">
    <rect width="110" id="el1000" fill="url(#lg1000)" transform="translate(294,128)" height="172" stroke="rgb(0,0,0)"/>
    <text text-anchor="middle" id="el1001" fill="rgb(0,0,0)" transform="translate(343,200)">50%</text>    
  </g>
  <g id="layer2">
    <rect width="110" id="el1002" fill="url(#lg1001)" transform="translate(494,128)" height="172" stroke="rgb(0,0,0)"/>
    <text text-anchor="middle" id="el1003" fill="rgb(0,0,0)" transform="translate(543,200)">100%</text>    
  </g>
</svg>
```

![ling](https://user-images.githubusercontent.com/3905787/30199088-bef49cf4-9479-11e7-94c5-506665920fa4.png)

**Comment by annulen** [comment]
Fixed in 3a09333e6f3383a694ba029678803f89216a69b9
